cli/src/semgrep/output.py

```diff
@@ -164,11 +214,13 @@ def __init__(
         self.engine_type: EngineType = EngineType.OSS
 
         self.final_error: Optional[Exception] = None
-        formatter_type = FORMATTERS.get(self.settings.output_format)
-        if formatter_type is None:
-            raise RuntimeError(f"Invalid output format: {self.settings.output_format}")
 
-        self.formatter = formatter_type()
+        self._formatters: Dict[Optional[str], BaseFormatter] = {}
```

Comment:
Actually, with the dict setup like this, is it possible that when we don't specify the output, but specify multiple formats, then will we only output to the last output format listed?

Reply:
No, I think you would just get one of the output formats. I wasn't thinking about this case because the CLI can only specify one output right now. I was planning on requiring that each format have a distinct destination. I will put a check in the normalize step to error if duplicate destinations where not given.

Reply:
Sounds good.